msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -132,6 +133,7 @@ public void onTaskCompleted(final List<ICacheRecord> result) {
                                 // We do not support switching from MULTIPLE to SINGLE.
                                 // See getAccountFromICacheRecordList() for more details.
                                 checkCurrentAccountNotifyCallback(callback, result);
+                                persistCurrentAccount(result);
```

Comment:
I believe we are persisting the same account that was already persisted?

Reply:
Yes, given that there might be any changes on the account object. 

An alternative would be to check everything inside the stored account and see if anything changes, which, IMHO, I think it's an overkill and might not be easily sustainable.